lib/screens/app/wallet/receive_confirmation.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart' hide Action;
+import 'package:flutter/rendering.dart';
+import 'package:seeds/models/cart_model.dart';
+import 'package:seeds/providers/services/navigation_service.dart';
+import 'package:seeds/widgets/cart_list_view.dart';
+import 'package:seeds/widgets/main_button.dart';
+import 'package:seeds/i18n/wallet.i18n.dart';
+
+class ReceiveConfirmation extends StatefulWidget {
+  final CartModel cart;
+
+  ReceiveConfirmation({Key key, @required this.cart}) : super(key: key);
+
+  @override
+  _ReceiveConfirmationState createState() => _ReceiveConfirmationState();
+}
+
+class _ReceiveConfirmationState extends State<ReceiveConfirmation> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomPadding: true,
+      appBar: AppBar(
+        leading: IconButton(
+          icon: Icon(Icons.arrow_back, color: Colors.black),
+          onPressed: () => Navigator.of(context).pop(),
+        ),
+        backgroundColor: Colors.transparent,
+        elevation: 0,
+        title: Text(
+          'Receipt',
+          style: TextStyle(color: Colors.black87),
+        ),
+      ),
+      backgroundColor: Colors.white,
+      body: Container(
+        margin: EdgeInsets.only(left: 15, right: 15),
+        child: Column(
+          children: [
+            Container(
+              child: CartListView(
+                cart: widget.cart,
+                onChange: () => setState(() {}),
```

Comment:
This is not Flutter Way. When we have several widgets that depend on the common state, we have to wrap them into InheritedWidget (or use Provider/Consumer abstraction in our case)

Reply:
InheritedWidget - interesting!

Reply:
Actually kinda disagree

The cart is an object and CartListView is a view that edits a cart object.

CartListView cannot depend on an outer widget having a cart - it should not know anything about any outer widgets. 

So in this case, this is perfect. 

CartListView can always display a cart, no matter what screen is on the outside, so that its encapsulating this view data and is an editor for that object.

Reply:
Flutter way < Good programming practices

Good programming practice in OO programming is decoupling whenever possible. Widgets that have to know specifics about widgets they're contained in sounds like a very bad idea.